api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java

```diff
@@ -403,18 +405,120 @@ public DbWorkspace setPublished(
       String workspaceNamespace, String firecloudName, boolean publish) {
     final DbWorkspace dbWorkspace = workspaceDao.getRequired(workspaceNamespace, firecloudName);
 
+    updateWorkspaceACL(
+        dbWorkspace.getWorkspaceNamespace(), dbWorkspace.getFirecloudName(), publish);
+
+    dbWorkspace.setPublished(publish);
+    return workspaceDao.saveWithLastModified(dbWorkspace, userProvider.get());
+  }
+
+  @Override
+  public void publishWorkspaceViaDB(
+      String workspaceNamespace, PublishWorkspaceRequest publishWorkspaceRequest) {
+    final DbWorkspace dbWorkspace = workspaceDao.getByNamespace(workspaceNamespace).orElseThrow();
+
+    Optional<DbFeaturedWorkspace> featuredWorkspace =
+        featuredWorkspaceDao.findByWorkspace(dbWorkspace);
+
+    DbFeaturedWorkspace dbFeaturedWorkspace;
+    if (featuredWorkspace.isPresent()) {
```

Comment:
`ifPresentOrElse` would work well here.  You can use it like:

`ws.ifPresentOrElse(dbWorkspace -> [do stuff if present], () -> [do stuff if not present])`